- [Assembly and annotation of Strain S4](#assembly-and-annotation-of-strain-s4)
  * [Spades Assembly using careful mode](#spades-assembly-using-careful-mode)
    + [Clean assembly](#clean-assembly)
  * [First Prokka Annotation](#first-prokka-annotation)
  * [Genome walking](#genome-walking)
    + [Make database](#make-database)
    + [Step1](#step1)
    + [Step2](#step2)
    + [Step3](#step3)
    + [Step4](#step4)
    + [Step5](#step5)
  * [Manually merge contigs](#manually-merge-contigs)
    + [uppercase](#uppercase)
    + [get link sequence](#get-link-sequence)
    + [delete and extract contig00018 and contig00021](#delete-and-extract-contig00018-and-contig00021)
    + [reverse-complement contig21](#reverse-complement-contig21)
    + [Singleline RC contig](#singleline-rc-contig)
    + [Singleline link](#singleline-link)
    + [merge all and send to new contig](#merge-all-and-send-to-new-contig)
    + [make sure its good](#make-sure-its-good)
  * [Still genome walking](#still-genome-walking)
    + [Step1b](#step1b)
    + [Step1b corrected](#step1b-corrected)
    + [Step2b](#step2b)
  * [Manually merge these contigs](#manually-merge-these-contigs)
    + [uppercase](#uppercase-1)
    + [get link sequence](#get-link-sequence-1)
    + [delete and extract contig00009 and contig00035](#delete-and-extract-contig00009-and-contig00035)
    + [singleline link](#singleline-link)
    + [merge all and send to new contig](#merge-all-and-send-to-new-contig-1)
    + [make sure its good](#make-sure-its-good-1)
    + [finalize](#finalize)
  * [Second prokka annotation](#second-prokka-annotation)

<small><i><a href='http://ecotrust-canada.github.io/markdown-toc/'>Table of contents generated with markdown-toc</a></i></small>



# Assembly and annotation of Strain S4

> After different preliminary analysis the idea was to assemble at high coverage Illumina Novaseq sequences, and since the biosynthetic cluster of interest was partial and terminated at a contigs end, we decide to manually blast this contigs end against the raw reads and see if possible to "walk" from these ends using a manual assembly tool (staden GAP4), even at  a much lower coverage than the original contigs.  Interestingly, some of the (150 bp) paired-end reads were in fact mate-pairs, and helped link some of these contigs.  

## Spades Assembly using careful mode

```
source activate prokka114
spades.py  --careful --pe1-1 /work/marcel/genome/00_RAW/S4_CGCTATGT-GTGTCGGA-BH2CJJDRXY_L002_R1.fastq --pe1-2 /work/marcel/genome/00_RAW/S4_CGCTATGT-GTGTCGGA-BH2CJJDRXY_L002_R2.fastq -o spades_fullset
```

list of packages installed in the miniconda and the state of the server [prokka114](https://github.com/suzumar/ROS4_manus/blob/main/files/other/prokka114.txt) environment  

[output](https://github.com/suzumar/ROS4_manus/blob/main/files/data/contigs.fasta) contigs.fasta file


### Clean assembly
run GC/coverage and use that and contig length to "clean" assembly

```
fastaoneline contigs.fasta
sed 's/NODE_/NODE/' contigs.fasta.fasta > S4.fasta
```

[fastaoneline](https://github.com/suzumar/ROS4_manus/blob/main/files/scripts/fastaoneline) script that formats fasta with entire sequence in the same line

remove contigs less than 500 bp (finish in line 16243)

```
cat S4.fasta | head -16244 > S4.long.fasta
cat S4.long.fasta | grep ">"  | gawk -F "_" '{printf("%s\t",$1);printf("%s\t",$3);printf("%s\n",$5)}'  > S4.1.tsv
cat S4.long.fasta |gawk -F "" '{if ($1 ~/>/) printf("%s\t", $0); else {i=1; j=0; while (i<=NF){if ($i ~ /G|C/) j=j+1;i=i+1}printf("%.4f\t",(j/(NF)));printf("%s\n",NF)}}'> S4.2.tsv
paste S4.1.tsv S4.2.tsv | sed 's/\>//' | awk '{print $1"\t"$2"\t"$3"\t"$5}' > S4.tsv
(echo 'node      length  coverage        gc' &&cat S4.tsv) > S4.3.tsv
sed "s/@p/S4/g" /usr/local/myscripts/for.bubble.R > /tmp/bubble.R
Rscript /tmp/bubble.R
```

> Note that there are tabs \<ctrl\>vi between node, length, coverage and gc.  

[for.bubble.R](https://github.com/suzumar/ROS4_manus/blob/main/files/scripts/for.bubble.R) a pre-script to make a bubble plot with GC content and coverage with length as the size of the bubble



Based on the graph I checked against NCBI :

```
Very High Coverage lower GC
NODE4078 : Streptomyces 16S rRNA
NODE6520 : Streptomyces 16S rRNA
NODE6528 : Streptomyces 23S rRNA
NODE3691 : Streptomyces 23S rRNA

Lower GC but > 1500 Coverage

NODE150 : Streptomyces
NODE278 : cloning vector
NODE3412 : beta lactamase cloning vector
NODE6907 : cloning vector

Lower coverage, GC>0.55

NODE140 : Streptomyces
NODE142 : Streptomyces
NODE397 : Streptomyces
NODE848 : Streptomyces

NODE1513 : Bosea 16S rRNA coverage 300
```

> So if coverage > 500 and GC > 0.55 likely Streptomyces
 

`cat S4.tsv | awk '{if(($3>500)&&($4>0.55))print $1"_length_"$2"_cov_"$3}' > S4_reads_to_keep`
`grep -w -A1 -f S4_reads_to_keep S4.fasta |sed '/--/d' > S4_pre_4Prokka.fasta`

final file

`cat S4_pre_4Prokka.fasta | awk '{if($1~/>/){printf("%s",">contig");printf("%05d\n",(NR-1)/2+1)}else {print $0}}' > S4_4Prokka.fasta`

## First Prokka Annotation

```
mkdir S4.ANNO
cd S4.ANNO/
source activate prokka114
uname -a > prokka.version
pip freeze >> prokka.version
conda list >> prokka.version
prokka --kingdom Bacteria --outdir out --locustag S4 ../S4.final.fasta
```

## Genome walking

### Make database
in /work/marcel/genome/00_RAW

create fasta file from fastq

`source activate bioinfo3`

list of packages installed in the miniconda [bioinfo3](https://github.com/suzumar/ROS4_manus/blob/main/files/other/bioinfo3.txt) environment  

`awk '(NR+3) % 4 == 0 || (NR+2) % 4 == 0' S4_CGCTATGT-GTGTCGGA-BH2CJJDRXY_L002_R1.fastq | sed 's/^@/>/' > S4R1.fasta`  

`awk '(NR+3) % 4 == 0 || (NR+2) % 4 == 0' S4_CGCTATGT-GTGTCGGA-BH2CJJDRXY_L002_R2.fastq | sed 's/^@/>/' > S4R2.fasta`  

`cat S4R* >S4raw`  

`makeblastdb -in S4raw -dbtype nucl`  

> Initially I made a "mistake" and assembled the wrong contigs : 18 and 21.  The contig from the assembly conataning the BGC of interest 8 in rthe merged assembly and 9 in the original assembly containing 18+21 : see below

### Step1  

in file  /work/marcel/rima/genomes/S4/spades_fullset/S4/S4.ANNO/out/PROKKA\_07152021.fna 

file [PROKKA_07152021.fna](https://github.com/suzumar/ROS4_manus/blob/main/files/data/PROKKA_07152021.fna)

Contig 18 : 3' 500 bp:

```
>contig00018
TCTGCGGCACCTCCCGCGGCCGCGCGCTGGTGACCGCACGGCAGATTGCCATGTACCTGTGCCGTGAGCTGACGGACCTGTCGCTGCCGAAGATCGGCGCGCTGTTCGGCGGGCGGGACCACACCACGGTGATGCACGCCGACCGGAAGATCCGCAATCTGATGGCCGAGCGGCGCTCCATCTACAACCAGGTCACCGAGCTCACGAACCGCATCAAGGCCGGCTGACGGCCCGACACCGCACGAGGGCGCCCCCGGAGGGATCCGGGGGCGCCCTTCTTCGTGTTCCCGCTCGCGCCAACTCCCAGGTGTGCGGACGACACGGCCCGCTCATGCACGTGCGGGGGGGTGCCTCTATGTCTTTCGTCAAGGAGGTCGGGGGCTTCTCGGCTGGTAGAAGGTTCCGTCGCGGAGCATGGCGAACAGGACGCTGATGCGGTGGCGGGCGAGGCGGAGGAGGGCCTGGGTGTGGGTTTTGCCGCGGGCTCTGCAGCGGTCGTA
```

```
contig00018_3pr
GACGCTGATGCGGTGGCGGGCGAGGCGGAGGAGGGCCTGGGTGTGGGTTTTGCCGCGGGCTCTGCAGCGGTCGTA
```

in /work/marcel/rima/genomes/S4/spades_fullset/S4/S4.ANNO/out

select reads which overlap contig00018_3pr at edges (pos 1 and pos 150)

`vi cont18_3pr.fasta`  
`blastn -db /work/marcel/genome/00_RAW/S4raw -query cont18_3pr.fasta -evalue 1e-30 -out cont18_3pr.blast  -outfmt 7`  

[file](https://github.com/suzumar/ROS4_manus/blob/main/files/data/cont18_3pr.fasta) file cont18_3pr.fasta

select reads which overlap contig00018_3pr at edges (pos 1 and pos 150)

`cat cont18_3pr.blast | awk '{if($9<2)print ">"$2}'| grep A00318 > cont18_3pr.names`  
`cat cont18_3pr.blast | awk '{if($9~150)print ">"$2}'>> cont18_3pr.names` 

retrieve reads from fasta files

`gawk 'FILENAME=="cont18_3pr.names" {a[$1]=$1} FILENAME=="/work/marcel/genome/00_RAW/S4raw.fasta" {if(a[$1]){print $0}}' cont18_3pr.names /work/marcel/genome/00_RAW/S4raw.fasta > retrieved_cont18_3pr`

`grep -f cont18_3pr.names retrieved_cont18_3pr |sort -k1 | awk '{printf("%s ",$1);printf("%s\n",$2);printf("%s\n",$3)}' > cont18_3pr_4gap4.fasta`

`mkdir /NetVolumes/data/Users/Shared/suzuki/rima_S4b/`  

this folder is shared with another server where gap4 resides  

```
server configuration
Darwin monera.obs-banyuls.fr 10.8.0 Darwin Kernel Version 10.8.0: Tue Jun  7 16:33:36 PDT 2011; root:xnu-1504.15.3~1/RELEASE_I386 i386
Staden 2002.0 and gap4 version 4.7 
```

`cp cont18_3pr_4gap4.fasta /NetVolumes/data/Users/Shared/suzuki/rima_S4b/`  
`fastasingleline /NetVolumes/data/Users/Shared/suzuki/rima_S4b/cont18_3pr_4gap4.fasta`  

script [fastasingleline](https://github.com/suzumar/ROS4_manus/blob/main/files/scripts/fastasingleline) makes a fasta file with name and sequence in the same line separate by a space

copy the 20K 3' of contig18 to that folder

`fastaoneline PROKKA_07152021.fna`  
`grep -A1 contig00018 PROKKA_07152021.fna.fasta | grep -v ">" > contig00018.fasta`  

`rangee 100789 120788 contig00018 PROKKA_07152021.fna.fasta | grep -v ">" > contig18_20K_3pr.fasta`  
`cp contig18_20K_3pr.fasta /NetVolumes/data/Users/Shared/suzuki/rima_S4b/`  


script [rangee](https://github.com/suzumar/ROS4_manus/blob/main/files/scripts/rangee) extracts a range of positions from $1 to $2 from sequence named $3 in file $4 

create single files containing each individual sequence, then a list of fasta files to be imported by gap4 

`cd  /NetVolumes/data/Users/Shared/suzuki/rima_S4b/`
`cat cont18_3pr_4gap4.fasta.fasta  | awk '{close(F);F=(NR)".fasta"}{print $3>>F}'`  
`ls [1-3]*.fasta > list1`  *note I made a mistake and only got a part of reads those from 40-99 missing*  
`echo contig18_20K_3pr.fasta >>list1`  

in gap 4 assemble using list1 as input

gap4 analysis

interesting in position 20067 there is an obvious SNP that actually might help to link the contig that might be linked to a SNP in 20091

> *contigs179+ -180, 137+ -138 proof linkage of C in 20067*

**other differences** downstream : 20288-9, 20301,20327-20328 9-31. -36, -143, -166, -105, -116, -178, -172) dont seem to link to snp above

149-150, 15-16, 14-13 139-140 seems to link to 180 and 138 above

161-162, 125-126, 127-128, 159-160, seems to link to the four read pairs above

101-102 35-36, 141-142 but 19-20 (SNIP 20067 SNIP20195)180-181 (SNIP20195) 

143-144 165-166 seems to link to the two read pairs above 293-294

105-106 115-116 177-178 171-172 but 129-130 131-132 173-174 133-134 33-34 167-168 (snip 20067) 39-?

it appears that the differences in **other differences**  above might be present downstream of contig since in 105-106 115-116 177-178 171-172 in addition in 35-36 and 31-32 none with the snip 20067

last one will be read172.  try to blast it against assembly

last one will be read172.  try to blast that against new assembly

```
mkdir 3pr1
mv [1-9]*.fasta 3pr1/
grep -B1 -f 3pr1/172.fasta cont18_3pr_4gap4.fasta > read172.fasta
cd /work/marcel/rima/genomes/S4/spades_fullset/S4/S4.ANNO/out
makeblastdb -in PROKKA_07152021.fna -out S4_newA -dbtype nucl
blastn -db S4_newA -query /NetVolumes/data/Users/Shared/suzuki/rima_S4b/read172.fasta -out read172.blast -evalue 10
```

overlaps not quite a small contig (contig00002) and not quite in beginning

### Step2

```
cd /work/marcel/rima/genomes/S4/spades_fullset/S4/S4.ANNO/out
rangee 76 150 A0 /NetVolumes/data/Users/Shared/suzuki/rima_S4b/read172.fasta > cont18b_3pr.fasta
blastn -db /work/marcel/genome/00_RAW/S4raw -query cont18b_3pr.fasta -evalue 1e-30 -out cont18b_3pr.blast  -outfmt 7 -num_alignments 10000
cat cont18b_3pr.blast | awk '{if($9<2)print ">"$2}'| grep A00318 > cont18b_3pr.names
cat cont18b_3pr.blast | awk '{if($9~150)print ">"$2}'>> cont18b_3pr.names
gawk 'FILENAME=="cont18b_3pr.names" {a[$1]=$1} FILENAME=="/work/marcel/genome/00_RAW/S4raw.fasta" {if(a[$1]){print $0}}' cont18b_3pr.names /work/marcel/genome/00_RAW/S4raw.fasta > retrieved_cont18b_3pr
grep -f cont18b_3pr.names retrieved_cont18b_3pr |sort -k1 | awk '{printf("%s ",$1);printf("%s\n",$2);printf("%s\n",$3)}' > cont18b_3pr_4gap4.fasta
cp cont18b_3pr_4gap4.fasta /NetVolumes/data/Users/Shared/suzuki/rima_S4b/
fastasingleline /NetVolumes/data/Users/Shared/suzuki/rima_S4b/cont18b_3pr_4gap4.fasta
```

```
cd  /NetVolumes/data/Users/Shared/suzuki/rima_S4b/
cat cont18b_3pr_4gap4.fasta.fasta  | awk '{close(F);F=(NR+184)".fasta"}{print $3>>F}'
ls [1-3]*.fasta > list2
```

In gap 4 assemble using list2 as input

last one will be read253. try to blast that against new assembly

```
grep -B1 -f 3pr2/253.fasta cont18b_3pr_4gap4.fasta > read253.fasta
blastn -db S4_newA -query /NetVolumes/data/Users/Shared/suzuki/rima_S4b/read253.fasta -out read235.blast -evalue 10
```

Overlaps with differences with contig4  

### Step3  

```
cd /work/marcel/rima/genomes/S4/spades_fullset/S4/S4.ANNO/out
rangee 1 75 A0 /NetVolumes/data/Users/Shared/suzuki/rima_S4b/read253.fasta > cont18c_3pr.fasta
blastn -db /work/marcel/genome/00_RAW/S4raw -query cont18c_3pr.fasta -evalue 1e-30 -out cont18c_3pr.blast  -outfmt 7 -num_alignments 10000
cat cont18c_3pr.blast | awk '{if($9<2)print ">"$2}'| grep A00318 > cont18c_3pr.names
cat cont18c_3pr.blast | awk '{if($9~150)print ">"$2}'>> cont18c_3pr.names
gawk 'FILENAME=="cont18c_3pr.names" {a[$1]=$1} FILENAME=="/work/marcel/genome/00_RAW/S4raw.fasta" {if(a[$1]){print $0}}' cont18c_3pr.names /work/marcel/genome/00_RAW/S4raw.fasta > retrieved_cont18c_3pr
grep -f cont18c_3pr.names retrieved_cont18c_3pr |sort -k1 | awk '{printf("%s ",$1);printf("%s\n",$2);printf("%s\n",$3)}' > cont18c_3pr_4gap4.fasta
cp cont18c_3pr_4gap4.fasta /NetVolumes/data/Users/Shared/suzuki/rima_S4b/
fastasingleline /NetVolumes/data/Users/Shared/suzuki/rima_S4b/cont18c_3pr_4gap4.fasta
```

```
cd  /NetVolumes/data/Users/Shared/suzuki/rima_S4b/
cat cont18c_3pr_4gap4.fasta.fasta  | awk '{close(F);F=(NR+308)".fasta"}{print $3>>F}'
ls [1-4]*.fasta > list3
```

read349 (RC) -read350 is the end  try to blast that against new assembly

```
mv [1-9]*.fasta 3pr3
grep -B1 -f 3pr3/349.fasta cont18c_3pr_4gap4.fasta > read349.fasta
blastn -db S4_newA -query /NetVolumes/data/Users/Shared/suzuki/rima_S4b/read349.fasta -out read349.blast -evalue 10
```

Hits with end of contig21 from assembly

```
cd /work/marcel/genome/S4/spades_fullset/S4/S4.ANNO/out
rangee 93423 113424 contig00021 PROKKA_07152021.fna.fasta | grep -v ">" > contig21_20K_3pr.fasta
cp contig21_20K_3pr.fasta /NetVolumes/data/Users/Shared/suzuki/rima_S4b/
```

### Step4 

```
cd /work/marcel/rima/genomes/S4/spades_fullset/S4/S4.ANNO/out
rangee 1 75 A0 /NetVolumes/data/Users/Shared/suzuki/rima_S4b/read349.fasta > cont18d_3pr.fasta
blastn -db /work/marcel/genome/00_RAW/S4raw -query cont18d_3pr.fasta -evalue 1e-30 -out cont18d_3pr.blast  -outfmt 7 -num_alignments 10000
cat cont18d_3pr.blast | awk '{if($9<2)print ">"$2}'| grep A00318 > cont18d_3pr.names
cat cont18d_3pr.blast | awk '{if($9~150)print ">"$2}'>> cont18d_3pr.names
gawk 'FILENAME=="cont18d_3pr.names" {a[$1]=$1} FILENAME=="/work/marcel/genome/00_RAW/S4raw.fasta" {if(a[$1]){print $0}}' cont18d_3pr.names /work/marcel/genome/00_RAW/S4raw.fasta > retrieved_cont18d_3pr
grep -f cont18d_3pr.names retrieved_cont18d_3pr |sort -k1 | awk '{printf("%s ",$1);printf("%s\n",$2);printf("%s\n",$3)}' > cont18d_3pr_4gap4.fasta
cp cont18d_3pr_4gap4.fasta /NetVolumes/data/Users/Shared/suzuki/rima_S4b/
fastasingleline /NetVolumes/data/Users/Shared/suzuki/rima_S4b/cont18d_3pr_4gap4.fasta
```
	
```
cd  /NetVolumes/data/Users/Shared/suzuki/rima_S4b/
cat cont18d_3pr_4gap4.fasta.fasta  | awk '{close(F);F=(NR+470)".fasta"}{print $3>>F}'
ls [1-9]*.fasta > list4
echo contig21_20K_3pr.fasta  >> list4
```

Try to blast the upstream region of contig21 and see if it fits 

### Step5

```
cd /work/marcel/rima/genomes/S4/spades_fullset/S4/S4.ANNO/out
vi cont21_3prMinus75.fasta
blastn -db /work/marcel/genome/00_RAW/S4raw -query cont21_3prMinus75.fasta -evalue 1e-30 -out cont21_3prMinus75.blast  -outfmt 7 -num_alignments 10000
cat cont21_3prMinus75.blast | awk '{if($9<2)print ">"$2}'| grep A00318 > cont21_3prMinus75.names
cat cont21_3prMinus75.blast | awk '{if($9~150)print ">"$2}'>> cont21_3prMinus75.names
gawk 'FILENAME=="cont21_3prMinus75.names" {a[$1]=$1} FILENAME=="/work/marcel/genome/00_RAW/S4raw.fasta" {if(a[$1]){print $0}}' cont21_3prMinus75.names /work/marcel/genome/00_RAW/S4raw.fasta > retrieved_cont21_3prMinus75
grep -f cont21_3prMinus75.names retrieved_cont21_3prMinus75 |sort -k1 | awk '{printf("%s ",$1);printf("%s\n",$2);printf("%s\n",$3)}' > cont21_3prMinus75_4gap4.fasta
cp cont21_3prMinus75_4gap4.fasta /NetVolumes/data/Users/Shared/suzuki/rima_S4b/
fastasingleline /NetVolumes/data/Users/Shared/suzuki/rima_S4b/cont21_3prMinus75_4gap4.fasta
```

```
cd  /NetVolumes/data/Users/Shared/suzuki/rima_S4b/
cat cont21_3prMinus75_4gap4.fasta.fasta  | awk '{close(F);F=(NR+614)".fasta"}{print $3>>F}'
ls [1-9]*.fasta > list5
```

links with **other differences**  
375-376 409-410  411-412 tha 657-658 555-556

weird some missing 300.fasta

It appears that those contigs are linked.  

## Manually merge contigs

make contig combining positions : 

contig18 then contig generated in gap4 post 20001-21193 then contig21 reverse complemented

513, 523 in cont21 interface

100, 283 in cont18 interface

in gap4 save contig as *S4_cons.fasta* 

`cd  /NetVolumes/data/Users/Shared/suzuki/rima_S4b/`   
`fastaoneline S4_cons.fasta`  

### uppercase

`cat S4_cons.fasta.fasta | sed '/contig/\!s/a/A/g;/contig/\!s/t/T/g;/contig/\!s/c/C/g;/contig/\!s/g/G/g' > S4_cons_final.fasta`  

### get link sequence

`cd /work/marcel/rima/genomes/S4/spades_fullset/S4/S4.ANNO/out`
`rangee 20001 21193 contig /NetVolumes/data/Users/Shared/suzuki/rima_S4b/S4_cons_final.fasta > ../../contig18_21_link.fasta`  

Under /work/marcel/rima/genomes/S4/spades_fullset/S4

### delete and extract contig00018 and contig00021

```
cd /work/marcel/rima/genomes/S4/spades_fullset/S4/
fastasingleline S4_4Prokka.fasta
grep contig00018 S4_4Prokka.fasta.fasta > contig00018.fastasl
grep contig00021 S4_4Prokka.fasta.fasta > contig00021.fastasl
grep contig00021 S4_4Prokka.fasta.fasta | awk '{print $1"\n"$2}' > contig00021.fasta
egrep -v 'contig00018|contig00021' S4_4Prokka.fasta.fasta > S4_4Prokka_minus18and21.fasta.fastasl
```

### reverse-complement contig21

`cat contig00021.fasta | awk 'BEGIN{RS=">";FS="\n";a["T"]="A";a["A"]="T";a["C"]="G";a["G"]="C";a["N"]="N"}NR>1{for (i=2;i<=NF;i++) seq=seq""$i;for(i=length(seq);i!=0;i--) {k=substr(seq,i,1);x=x a[k]}; printf ">%s\n%s",$1,x}' > contig00021rc.fasta`   

### Singleline RC contig

`fastasingleline contig00021rc.fasta`   

### Singleline link

`fastasingleline contig18_21_link.fasta`

### merge all and send to new contig

`paste contig00018.fastasl contig18_21_link.fasta.fasta contig00021rc.fasta.fasta |awk '{printf("%s ",$1);printf("%s",$2);printf("%s",$5);printf("%s\n", $7)}' | awk '{print length($2)}'`  

> 235405

`paste contig00018.fastasl contig18_21_link.fasta.fasta contig00021rc.fasta.fasta |awk '{printf("%s ",$1);printf("%s",$2);printf("%s",$5);printf("%s\n", $7)}' > contig18_21_merged.fastasl`  

### make sure its good 

```
cat contig18_21_merged.fastasl | awk '{print $1"\n"$2}' > contig18_21_merged.fasta
makeblastdb -in contig18_21_merged.fasta -out /tmp/cont1821 -dbtype nucl
blastn -query /NetVolumes/data/Users/Shared/suzuki/rima_S4b/3pr1/100.fasta -db /tmp/cont1821
blastn -query /NetVolumes/data/Users/Shared/suzuki/rima_S4b/3pr2/283.fasta -db /tmp/cont1821
blastn -query /NetVolumes/data/Users/Shared/suzuki/rima_S4b/3pr4/513.fasta -db /tmp/cont1821
blastn -query /NetVolumes/data/Users/Shared/suzuki/rima_S4b/3pr4/523.fasta -db /tmp/cont1821
cat contig18_21_merged.fastasl S4_4Prokka_minus18and21.fastasl | grep -c ">"
```

130

finalize 

`cat contig18_21_merged.fastasl S4_4Prokka_minus18and21.fastasl | awk '{print $1" "length($2)" "$2}'| sort -n -r -k 2 | awk '{printf("%s",">contig");printf("%05d\n",NR);printf("%s\n",$3)}' > S4_manu.fasta`  

> An antismash analysis indicated that I actually merged the "wrong" contigs since they do not include the angucycline BGC, so I continued the genome walk, now with starting with contig00009

## Still genome walking

under /work/marcel/rima/genomes/S4/spades_fullset/S4

```
mkdir S4_manu.WALK
cp S4_manu.fasta S4_manu.WALK/
vi contig9_3pr.fasta
makeblastdb -in S4_manu.fasta -out S4_manu -dbtype nucl
```

[file](https://github.com/suzumar/ROS4_manus/blob/main/files/data/cont9_3pr.fasta) contig9_3pr.fasta file

### Step1b

```
cd /work/marcel/rima/genomes/S4/spades_fullset/S4/S4_manu.WALK
vi cont9_3pr.fasta
blastn -db /work/marcel/genome/00_RAW/S4raw -query cont9_3pr.fasta -evalue 1e-30 -out cont9_3pr.blast  -outfmt 7  
cat cont9_3pr.blast | awk '{if($9<15)print ">"$2}'| grep A00318 > cont9_3pr.names
cat cont9_3pr.blast | awk '{if($9>145)print ">"$2}'>> cont9_3pr.names
gawk 'FILENAME=="cont9_3pr.names" {a[$1]=$1} FILENAME=="/work/marcel/genome/00_RAW/S4raw.fasta" {if(a[$1]){print $0}}' cont9_3pr.names /work/marcel/genome/00_RAW/S4raw.fasta > retrieved_cont9_3pr
grep -f cont9_3pr.names retrieved_cont9_3pr |sort -k1 | awk '{printf("%s ",$1);printf("%s\n",$2);printf("%s\n",$3)}' > cont9_3pr_4gap4.fasta
mkdir /NetVolumes/data/Users/Shared/suzuki/rima_s4c/
cp cont9_3pr_4gap4.fasta /NetVolumes/data/Users/Shared/suzuki/rima_s4c/
fastasingleline /NetVolumes/data/Users/Shared/suzuki/rima_S4c/cont9_3pr_4gap4.fasta
```

```
rangee 191624 211623 contig00009 S4_manu.fasta | grep -v ">" > contig9_20K_3pr.fasta
cp contig9_20K_3pr.fasta /NetVolumes/data/Users/Shared/suzuki/rima_s4c/
cd /NetVolumes/data/Users/Shared/suzuki/rima_S4c/
cat cont9_3pr_4gap4.fasta.fasta  | awk '{close(F);F=(NR)".fasta"}{print $3>>F}'
ls [1-9]*.fasta > list1
echo contig9_20K_3pr.fasta >>list1 
mv [1-9]*.fasta 3pr1
```

assemble with GAP4
read33 new3 3pr

`
cd /work/marcel/rima/genomes/S4/spades_fullset/S4/S4_manu.WALK
blastn -db S4_manu -query /NetVolumes/data/Users/Shared/suzuki/rima_S4c/3pr1/33.fasta -out read33.blast -evalue 10`  

no hits

```
grep -B1 -f /NetVolumes/data/Users/Shared/suzuki/rima_S4c/3pr1/33.fasta cont9_3pr_4gap4.fasta > read33.fasta
rangee 1 75 A00318 read33.fasta > cont9b_3pr.fasta
```  

### Step1b corrected  

add more (up to 10,000) reads to output

```
blastn -db /work/marcel/genome/00_RAW/S4raw -query cont9_3pr.fasta -evalue 1e-30 -out cont9_3pr.blast  -outfmt 7  -num_alignments 10000
cat cont9_3pr.blast | awk '{if($9<15)print ">"$2}'| grep A00318 > cont9z_3pr.names
cat cont9_3pr.blast | awk '{if($9>145)print ">"$2}'>> cont9z_3pr.names
mv cont9z_3pr.names cont9w_3pr_names
grep -v -f cont9_3pr.names cont9w_3pr_names > cont9z_3pr_names
gawk 'FILENAME=="cont9z_3pr.names" {a[$1]=$1} FILENAME=="/work/marcel/genome/00_RAW/S4raw.fasta" {if(a[$1]){print $0}}' cont9z_3pr.names /work/marcel/genome/00_RAW/S4raw.fasta > retrieved_cont9z_3pr
grep -f cont9z_3pr.names retrieved_cont9z_3pr |sort -k1 | awk '{printf("%s ",$1);printf("%s\n",$2);printf("%s\n",$3)}' > cont9z_3pr_4gap4.fasta
cp cont9z_3pr_4gap4.fasta /NetVolumes/data/Users/Shared/suzuki/rima_S4c/
fastasingleline /NetVolumes/data/Users/Shared/suzuki/rima_S4c/cont9z_3pr_4gap4.fasta
```

```
cd /NetVolumes/data/Users/Shared/suzuki/rima_S4c/
cat cont9z_3pr_4gap4.fasta.fasta  | awk '{close(F);F=(NR+40)".fasta"}{print $3>>F}'
ls [1-9]*.fasta > list2
mv [1-9]*.fasta 3pr1
```

read33 still new3 3pr

### Step2b  
  
```
cd /work/marcel/rima/genomes/S4/spades_fullset/S4/S4_manu.WALK
blastn -db /work/marcel/genome/00_RAW/S4raw -query cont9b_3pr.fasta -evalue 1e-30 -out cont9b_3pr.blast  -outfmt 7
```

no hits!
try the entire read33 still bad.  try removing filter dust

```	
blastn -db /work/marcel/genome/00_RAW/S4raw -query cont9b_3pr.fasta -dust no -evalue 1e-30 -out cont9b_3pr.blast  -num_alignments 10000 -outfmt 7
cat cont9b_3pr.blast | awk '{if($9<5)print ">"$2}'| grep A00318 > cont9b_3pr.names
cat cont9b_3pr.blast | awk '{if($9>100)print ">"$2}'>> cont9b_3pr.names
gawk 'FILENAME=="cont9b_3pr.names" {a[$1]=$1} FILENAME=="/work/marcel/genome/00_RAW/S4raw.fasta" {if(a[$1]){print $0}}' cont9b_3pr.names /work/marcel/genome/00_RAW/S4raw.fasta > retrieved_cont9b_3pr
grep -f cont9b_3pr.names retrieved_cont9b_3pr |sort -k1 | awk '{printf("%s ",$1);printf("%s\n",$2);printf("%s\n",$3)}' > cont9b_3pr_4gap4.fasta
cp cont9b_3pr_4gap4.fasta /NetVolumes/data/Users/Shared/suzuki/rima_S4c/
fastasingleline /NetVolumes/data/Users/Shared/suzuki/rima_S4c/cont9b_3pr_4gap4.fasta
```

```
cd  /NetVolumes/data/Users/Shared/suzuki/rima_S4c/
cat cont9b_3pr_4gap4.fasta.fasta  | awk '{close(F);F=(NR+94)".fasta"}{print $3>>F}'
ls [1-9]*.fasta > list3
mkdir 3pr3
mv [1-9]*.fasta 3pr3
```

read104 new end  

```
cd /work/marcel/rima/genomes/S4/spades_fullset/S4/S4_manu.WALK
grep -B1 -f /NetVolumes/data/Users/Shared/suzuki/rima_S4c/3pr3/104.fasta cont9_3pr_4gap4.fasta > read104.fasta
blastn -db S4_manu -query /NetVolumes/data/Users/Shared/suzuki/rima_S4c/3pr1/104.fasta -out read104.blast -evalue 10
```

hits with Contig00035 5'  

```
rangee 1 20000 contig00035 S4_manu.fasta > contig35_30K_3pr.fasta
echo contig35_30K_3pr.fasta > list4
```

## Manually merge these contigs 

`cd /work/marcel/genome/S4/spades_fullset/S4/S4_manu.WALK`


make contig combining positions : 

contig9 then contig generated in gap4 post 20001-20086 then contig35

23 35 in cont9 interface

103, 146 in cont35 interface

save contig as *S4_cons2.fasta* 

`fastaoneline S4_cons2.fasta`  

### uppercase

`cat S4_cons2.fasta.fasta | sed '/contig/\!s/a/A/g;/contig/\!s/t/T/g;/contig/\!s/c/C/g;/contig/\!s/g/G/g' > S4_cons2_final.fasta`  

### get link sequence

`rangee 20001 20086 contig /NetVolumes/data/Users/Shared/suzuki/rima_S4c/S4_cons2_final.fasta > contig9_35_link.fasta`  

under /work/marcel/genome/S4/spades_fullset/S4

### delete and extract contig00009 and contig00035

```
fastasingleline S4_manu.fasta
grep contig00009 S4_manu.fasta.fasta > contig00009.fastasl
grep contig00035 S4_manu.fasta.fasta > contig00035.fastasl
egrep -v 'contig00009|contig00035' S4_manu.fasta.fasta > S4_manu_minus9and35.fasta.fastasl
```

### singleline link

`fastasingleline contig9_35_link.fasta`

### merge all and send to new contig

`paste contig00009.fastasl contig9_35_link.fasta.fasta contig00035.fastasl |awk '{printf("%s ",$1);printf("%s",$2);printf("%s",$5);printf("%s\n", $7)}' | awk '{print length($2)}'`  

> 289029

`paste contig00009.fastasl contig9_35_link.fasta.fasta contig00035.fastasl |awk '{printf("%s ",$1);printf("%s",$2);printf("%s",$5);printf("%s\n", $7)}' > contig9_35_merged.fastasl`  

### make sure its good 

```
cat contig9_35_merged.fastasl | awk '{print $1"\n"$2}' > contig9_35_merged.fasta
makeblastdb -in contig9_35_merged.fasta -out /tmp/cont935 -dbtype nucl
blastn -query /NetVolumes/data/Users/Shared/suzuki/rima_S4c/3pr1/23.fasta -db /tmp/cont935
blastn -query /NetVolumes/data/Users/Shared/suzuki/rima_S4c/3pr1/35.fasta -db /tmp/cont935
blastn -query /NetVolumes/data/Users/Shared/suzuki/rima_S4c/3pr2/103.fasta -db /tmp/cont935
blastn -query /NetVolumes/data/Users/Shared/suzuki/rima_S4c/3pr2/146.fasta -db /tmp/cont935
cat contig9_35_merged.fastasl S4_manu_minus9and35.fasta.fastasl | grep -c ">"
```  

130

### finalize 

`cat contig9_35_merged.fastasl S4_manu_minus9and35.fasta.fastasl | awk '{print $1" "length($2)" "$2}'| sort -n -r -k 2 | awk '{printf("%s",">contig");printf("%05d\n",NR);printf("%s\n",$3)}' > S4_manu2.fasta` 

## Second prokka annotation

in /work/marcel/genome/S4/spades\_fullset/S4/S4_manu.WALK  
	
```
mkdir S4.ANNO
cd S4.ANNO/
source activate prokka114
uname -a > prokka.version
pip freeze >> prokka.version
conda list >> prokka.version
prokka --kingdom Bacteria --outdir out --locustag S4 ../S4_manu2.fasta
```

repeat with option --cdsrnaolap

```
prokka --compliant --centre LBBM -kingdom Bacteria --cdsrnaolap --outdir PRJNA785135_noCDSrna  --locustag ROS4 --prefix STRPM-ROS4 --addgenes --force --gffve 2 --genus Streptomyces --species sp. --strain RO-S4 --gcode 11  --usegenus  --proteins GCF_018638305.1_ASM1863830v1_genomic.gbk ../S4_manu3.fasta
```

## Curation at Genbank Submission Stage

genebank check indicated that some adapters were still there after annotation.  I will have to remove them "by hand" They are mostly at the ends of contigs.

```
Trim:
Sequence name, length, span(s), apparent source
ROS4_10	202023	1..109	adaptor:NGB01086.1
ROS4_103	6039	1..38	adaptor:NGB00754.1
ROS4_107	5199	1..69	adaptor:NGB01086.1
ROS4_2	289029	288956..289029	adaptor:NGB01086.1
ROS4_23	106470	106395..106470	adaptor:NGB01086.1
ROS4_27	93376	93357..93376	adaptor:NGB01088.1
ROS4_29	86658	1..103	adaptor:NGB01086.1
ROS4_38	67533	1..54	adaptor:NGB01086.1
ROS4_39	66013	65939..66013	adaptor:NGB01086.1
ROS4_51	43600	43474..43543	adaptor:NGB01086.1
ROS4_55	40816	1..51	adaptor:NGB00754.1
ROS4_68	30058	1..54	adaptor:NGB00848.1
ROS4_69	29620	1..59	adaptor:NGB00848.1
ROS4_81	19305	1..75	adaptor:NGB01086.1
ROS4_82	17849	1..54	adaptor:NGB00848.1
ROS4_85	15272	15199..15272	adaptor:NGB01086.1
```

the contig numbers are those is the manually assembled genome

`/work/marcel/rima/genomes/S4/spades_fullset/S4/S4_manu.WALK/S4_manu.fasta`  

afterwards try to reassembles with minimus2 to see if any now overlaps

### Cleaning contaminant ends

saved errors in 

`/work/marcel/rima/genomes/S4/spades_fullset/S4/S4_manu.WALK`  

```
vi nbci_contamin.txt
cat nbci_contamin.txt | sort -nk 3 |head -11 | tail -10 > ncbi_conta_5pr.txt
cat nbci_contamin.txt | sort -nk 3 | tail -6 > ncbi_conta_3pr.txt

cat ncbi_conta_5pr.txt | awk '{if(length($1)==6){gsub("ROS4_","contig0000");print}else{if(length($1)==7){gsub("ROS4_","contig000");print}else{if(length($1)==8){gsub("ROS4_","contig00");print}}}}' | awk '{gsub("\\.\\.","\t");print}' | awk '{print "/"$1"/s/ .\\{"$4"\\}/ /"}' > contam_5pr.sed

cat ncbi_conta_3pr.txt | awk '{if(length($1)==6){gsub("ROS4_","contig0000");print}else{if(length($1)==7){gsub("ROS4_","contig000");print}else{if(length($1)==8){gsub("ROS4_","contig00");print}}}}' | awk '{gsub("\\.\\.","\t");print}'| awk '{print "/"$1"/s/.\\{"$2-$3+1"\\}$//"}' > contam_3pr.sed

fastasingleline S4_manu2.fasta

cat S4_manu2.fasta.fasta | sed -f contam_5pr.sed | sed -f contam_3pr.sed | tr ' ' '\012' > S4_manu3.fasta

prokka --compliant --centre LBBM -kingdom Bacteria --cdsrnaolap --outdir PRJNA785135  --locustag ROS4 --prefix STRPM-ROS4 --addgenes --force --gffve 2 --genus Streptomyces --species sp. --strain RO-S4 --gcode 11  --usegenus  --proteins GCF_018638305.1_ASM1863830v1_genomic.gbk ../S4_manu3.fasta

```

[file](https://github.com/suzumar/ROS4_manus/blob/main/files/other/nbci_contamin.txt) ncbi_contamin.txt

will need to remove : Partial rRNA from 

```
FATAL: RRNA_NAME_CONFLICTS: 3 rRNA product names are not standard. Correct the names to the standard format, eg "16S ribosomal RNA"
ORIG/PRJNA785135.sqn:rRNA	23S ribosomal RNA (partial)	gnl|LBBM|ROS4_17:c125137-123748	ROS4_03364
ORIG/PRJNA785135.sqn:rRNA	16S ribosomal RNA (partial)	gnl|LBBM|ROS4_119:c979-8	ROS4_06822
ORIG/PRJNA785135.sqn:rRNA	16S ribosomal RNA (partial)	gnl|LBBM|ROS4_123:c609-93	ROS4_06825
SUSPECT_RRNA_PRODUCTS: 3 rRNA product names contain suspect phrase
SUSPECT_RRNA_PRODUCTS: 3 rRNA product names contains 'partial'
ORIG/PRJNA785135.sqn:rRNA	23S ribosomal RNA (partial)	gnl|LBBM|ROS4_17:c125137-123748	ROS4_03364
ORIG/PRJNA785135.sqn:rRNA	16S ribosomal RNA (partial)	gnl|LBBM|ROS4_119:c979-8	ROS4_06822
ORIG/PRJNA785135.sqn:rRNA	16S ribosomal RNA (partial)	gnl|LBBM|ROS4_123:c609-93	ROS4_06825
FATAL: SHORT_RRNA: 3 rRNA features are too short
ORIG/PRJNA785135.sqn:rRNA	23S ribosomal RNA (partial)	gnl|LBBM|ROS4_17:c125137-123748	ROS4_03364
ORIG/PRJNA785135.sqn:rRNA	16S ribosomal RNA (partial)	gnl|LBBM|ROS4_119:c979-8	ROS4_06822
ORIG/PRJNA785135.sqn:rRNA	16S ribosomal RNA (partial)	gnl|LBBM|ROS4_123:c609-93	ROS4_06825
```

try to fix directly the .tbl file 

### remove partial rRNA genes

`in /work/marcel/rima/genomes/S4/spades_fullset/S4/S4_manu.WALK/S4.ANNO/PRJNA785135/STRPM-ROS4.tbl`


> lines 24483 24489

```
125137  123748  gene
                        locus_tag       ROS4_03351
125137  123748  rRNA
                        locus_tag       ROS4_03351
                        note    aligned only 43 percent of the 23S ribosomal RNA
                        product 23S ribosomal RNA (partial)
                        score   0
```

> lines 50183 50189 

```
979     8       gene
                        locus_tag       ROS4_06798
979     8       rRNA
                        locus_tag       ROS4_06798
                        note    aligned only 61 percent of the 16S ribosomal RNA
                        product 16S ribosomal RNA (partial)
                        score   7.7e-257
```

> lines 50208 50214

```
609     93      gene
                        locus_tag       ROS4_06801
609     93      rRNA
                        locus_tag       ROS4_06801
                        note    aligned only 32 percent of the 16S ribosomal RNA
                        product 16S ribosomal RNA (partial)
                        score   1.5e-142
```


```
cat  STRPM-ROS4.tbl | sed '24483,24489d;50183,50189d;50208,50214d' >  STRPM-ROS4_corr1.tbl
```

### Still errots

Need to remove --cdsrnaolap and sdd annotation of UrdM-like ORF  

so it shows up in the Genome

`in /work/marcel/rima/genomes/S4/spades_fullset/S4/S4_manu.WALK/S4.ANNO/PRJNA785135_noCDSrna/STRPM-ROS4.tbl`

> lines 3620 3626


```
193820  195808  gene
                        locus_tag       ROS4_00495
193820  195808  CDS
                        inference       ab initio prediction:Prodigal:002006
                        locus_tag       ROS4_00495
                        product SDR family oxidoreductase
                        protein_id      gnl|LBBM|ROS4_00495
```

`in /work/marcel/rima/genomes/S4/spades_fullset/S4/S4_manu.WALK/S4.ANNO/PRJNA785135/STRPM-ROS4_corr1.tbl`


>lines 3599

```
194462  194387  gene
                        locus_tag       ROS4_00492
194462  194387  tRNA
                        inference       COORDINATES:profile:Aragorn:001002
                        locus_tag       ROS4_00492
                        product tRNA-Ala(ggc)
```

> So the idea will be 

- to use the same locus number ROS_00492 in the patch lines 3620-26 that will be included, 
- to insert it after line 3598 of STRPM-ROS4\_corr1.tbl, that will end at 3605
- then delete lines 3606-3611

in /work/marcel/rima/genomes/S4/spades_fullset/S4/S4_manu.WALK/S4.ANNO/PRJNA785135/


```
cat ../PRJNA785135_noCDSrna/STRPM-ROS4.tbl | sed -n '3620,3626p' |sed 's/00495/00492/g' > patch.txt
cat patch.txt | sed '3598r /dev/stdin' STRPM-ROS4_corr1.tbl | sed '3606,3611d' > STRPM-ROS4_corr2.tbl
```

[file](https://github.com/suzumar/ROS4_manus/blob/main/files/other/patch.txt) patch.txt

### Fix locus tags, and latest errors in submission 

see error [file](https://github.com/suzumar/ROS4_manus/blob/main/files/other/errors.txt) : 

to account for 3 regions deleted above, plus officially given locus tag LRR80

`in /work/marcel/rima/genomes/S4/spades_fullset/S4/S4_manu.WALK/S4.ANNO/PRJNA785135/`

```
cat STRPM-ROS4_corr2.tbl | grep locus_tag  |sort|uniq | awk '{printf("s/%s/",$2);printf("%s","LRR80_");printf("%05d/\n",NR)}' > corr.sed
sed -f corr.sed STRPM-ROS4_corr2.tbl | sed $'s/\;/\;\t\t\tnote\t/' | tr ';' '\012' | sed '/ISfinder/s/inference/note/1' > STRPM-ROS4_final.tbl
```

### Make final files

```
cd /work/marcel/rima/genomes/S4/spades_fullset/S4/S4_manu.WALK/S4.ANNO/PRJNA785135/
mkdir corrected
cp STRPM-ROS4_final.tbl corrected
cp STRPM-ROS4.fsa corrected/STRPM-ROS4_final.fsa
cd corrected
tbl2asn -t STRPM-ROS4_final.sbt -V b -a r10k -l paired-ends -M n -N 1 -y 'Annotated using prokka 1.14.6 from https://github.com/tseemann/prokka' -Z STRPM\-ROS4\_final\.err -i STRPM\-ROS4\_final\.fsa
```
errors:
```
[tbl2asn 25.8] ERROR: Unknown qualifier 'score', relative line 24499
[tbl2asn 25.8] ERROR: Unknown qualifier 'score', relative line 50302
```

### Review of the assembly

in /work/marcel/rima/genomes/S4/spades_fullset/S4/S4_manu.WALK/S4.ANNO/PRJNA785135/corrected

	
```
source activate gass
gaas_fasta_statistics.pl -f STRPM-ROS4_final.fsa -o summary.txt
```	

[gass](https://github.com/suzumar/ROS4_manus/blob/main/files/other/gass.txt) environment  
[file](https://github.com/suzumar/ROS4_manus/blob/main/files/other/STRPM-ROS4_final_stat.txt) summary of assembly
